storage/garbage.go

```diff
@@ -0,0 +1,138 @@
+)
+
+func (m *Miner) storeGarbage(ctx context.Context, sectorID uint64, existingPieceSizes []uint64, sizes ...uint64) ([]Piece, error) {
+	if len(sizes) == 0 {
+		return nil, nil
+	}
+
+	deals := make([]actors.StorageDeal, len(sizes))
+	for i, size := range sizes {
+		commP, err := sectorbuilder.GeneratePieceCommitment(io.LimitReader(rand.New(rand.NewSource(42)), int64(size)), size)
+		if err != nil {
+			return nil, err
+		}
+
+		sdp := actors.StorageDealProposal{
+			PieceRef:             commP[:],
+			PieceSize:            size,
+			PieceSerialization:   actors.SerializationUnixFSv0,
+			Client:               m.worker,
+			Provider:             m.maddr,
+			ProposalExpiration:   math.MaxUint64,
+			Duration:             math.MaxUint64 / 2, // /2 because overflows
+			StoragePricePerEpoch: types.NewInt(0),
+			StorageCollateral:    types.NewInt(0),
+			ProposerSignature:    nil,
+		}
+
+		if err := api.SignWith(ctx, m.api.WalletSign, m.worker, &sdp); err != nil {
+			return nil, xerrors.Errorf("signing storage deal failed: ", err)
+		}
+
+		storageDeal := actors.StorageDeal{
+			Proposal: sdp,
+		}
+		if err := api.SignWith(ctx, m.api.WalletSign, m.worker, &storageDeal); err != nil {
+			return nil, xerrors.Errorf("signing storage deal failed: ", err)
+		}
+
+		deals[i] = storageDeal
+	}
+
+	params, aerr := actors.SerializeParams(&actors.PublishStorageDealsParams{
+		Deals: deals,
+	})
+	if aerr != nil {
+		return nil, xerrors.Errorf("serializing PublishStorageDeals params failed: ", aerr)
+	}
+
+	// TODO: We may want this to happen after fetching data
+	smsg, err := m.api.MpoolPushMessage(ctx, &types.Message{
+		To:       actors.StorageMarketAddress,
+		From:     m.worker,
+		Value:    types.NewInt(0),
+		GasPrice: types.NewInt(0),
+		GasLimit: types.NewInt(1000000),
+		Method:   actors.SMAMethods.PublishStorageDeals,
+		Params:   params,
+	})
+	if err != nil {
+		return nil, err
+	}
+	r, err := m.api.StateWaitMsg(ctx, smsg.Cid())
+	if err != nil {
+		return nil, err
+	}
+	if r.Receipt.ExitCode != 0 {
+		log.Error(xerrors.Errorf("publishing deal failed: exit %d", r.Receipt.ExitCode))
+	}
+	var resp actors.PublishStorageDealResponse
+	if err := resp.UnmarshalCBOR(bytes.NewReader(r.Receipt.Return)); err != nil {
+		return nil, err
+	}
+	if len(resp.DealIDs) != len(sizes) {
+		return nil, xerrors.New("got unexpected number of DealIDs from PublishStorageDeals")
+	}
+
+	out := make([]Piece, len(sizes))
+
+	for i, size := range sizes {
+		name := fmt.Sprintf("fake-file-%d", rand.Intn(100000000))
+		ppi, err := m.sb.AddPiece(size, sectorID, io.LimitReader(rand.New(rand.NewSource(42)), int64(size)), existingPieceSizes)
+		if err != nil {
+			return nil, err
+		}
+
+		existingPieceSizes = append(existingPieceSizes, size)
+
+		out[i] = Piece{
+			DealID: resp.DealIDs[i],
+			Ref:    name,
+			Size:   ppi.Size,
+			CommP:  ppi.CommP[:],
+		}
+	}
+
+	return out, nil
+}
+
+func (m *Miner) StoreGarbageData(_ context.Context) error {
+	ctx := context.TODO()
```

Comment:
context.TODO, and ignoring the context in the function parameter?

Reply:
The function ctx gets cancelled when the cmd returns

Reply:
oh, thats kinda annoying...

Reply:
probably should add a comment letting future us know that